### PR TITLE
Add tests for v1.4.1 contracts

### PR DESCRIPTION
**Pull request by Uxio0**
- Add v1.4.1 test cases
- Refactor tests/code for v1.4.1
- Refactor tests to keep v1.3.0 and previous tests running

### CODE REVIEW

File: gnosis/safe/safe_creator.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    @staticmethod
+    def deploy_compatibility_fallback_handler(
+        ethereum_client: EthereumClient, deployer_account: LocalAccount
+    ) -> EthereumTxSent:
+        """
+        Deploy Last compatibility Fallback handler
+
+        :param ethereum_client:
+        :param deployer_account: Ethereum account
+        :return: deployed contract address
```
Comment (moisses89) [inline gnosis/safe/safe_creator.py:117]: Maybe better EthereumTxSent with the contract address.

File: gnosis/safe/safe_creator.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    @staticmethod
+    def deploy_simulate_tx_accessor(
+        ethereum_client: EthereumClient, deployer_account: LocalAccount
+    ) -> EthereumTxSent:
+        """
+        Deploy Last compatibility Fallback handler
+
+        :param ethereum_client:
+        :param deployer_account: Ethereum account
+        :return: deployed contract address
```
Comment (moisses89) [inline gnosis/safe/safe_creator.py:144]: Same as above

File: gnosis/safe/safe_creator.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            ethereum_tx_sent.contract_address,
+            deployer_account.address,
+        )
+        return ethereum_tx_sent
+
+    @staticmethod
+    def estimate_safe_creation_2(
+        ethereum_client: EthereumClient,
+        master_copy_address: str,
+        proxy_factory_address: str,
```
Comment (moisses89) [inline gnosis/safe/safe_creator.py:166]: Better ChecksumAddress instead str

File: gnosis/safe/safe_creator.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        number_owners: int,
+        gas_price: int,
+        payment_token: Optional[str],
+        payment_receiver: str = NULL_ADDRESS,
+        fallback_handler: Optional[str] = None,
+        payment_token_eth_value: float = 1.0,
+        fixed_creation_cost: Optional[int] = None,
+    ) -> SafeCreationEstimate:
```
Comment (moisses89) [inline gnosis/safe/safe_creator.py:174]: I would like to have documented this function also.

File: gnosis/safe/safe_creator.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+            safe_creation_tx.gas_price,
+            safe_creation_tx.payment,
+            safe_creation_tx.payment_token,
+        )
+
+    @staticmethod
+    def build_safe_create2_tx(
+        ethereum_client: EthereumClient,
+        master_copy_address: str,
+        proxy_factory_address: str,
```
Comment (moisses89) [inline gnosis/safe/safe_creator.py:208]: ChecksumAddress

File: gnosis/safe/tests/test_safe_create2_tx.py [hunk 6]
```diff
@@ -64,6 +64,8 @@ def test_safe_create2_tx_builder(self):
         logs = self.proxy_factory_contract.events.ProxyCreation().process_receipt(
             tx_receipt
         )
+        print(logs)
+        print(tx_receipt.logs)
```
Comment (moisses89) [inline gnosis/safe/tests/test_safe_create2_tx.py:68]: Missing prints

File: gnosis/safe/safe.py [hunk 7]
```diff
@@ -75,7 +67,7 @@ class SafeInfo:
     version: str
 
 
-class SafeBase(ContractBase, metaclass=ABCMeta):
+class Safe(SafeCreator, ContractBase, metaclass=ABCMeta):
```
Comment (moisses89) [inline gnosis/safe/safe.py:70]: Why Safe extends SafeCreator?

**Review by moisses89 — CHANGES_REQUESTED** [review]

File: gnosis/eth/contracts/__init__.py [hunk 8]
```diff
@@ -149,6 +153,17 @@ def get_safe_V1_4_1_contract(w3: Web3, address: Optional[str] = None) -> Contrac
     pass
 
 
+def get_compatibility_fallback_handler_contract(
+    w3: Web3, address: Optional[ChecksumAddress] = None
+) -> Contract:
+    """
+    :param w3:
+    :param address:
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply gnosis/eth/contracts/__init__.py:161]: It doesn't need to be a Safe, but makes sense

File: gnosis/safe/multi_send.py [hunk 9]
```diff
@@ -306,4 +310,4 @@ def build_tx_data(self, multi_send_txs: List[MultiSendTx]) -> bytes:
         encoded_multisend_data = b"".join([x.encoded_data for x in multi_send_txs])
         return multisend_contract.functions.multiSend(
             encoded_multisend_data
-        ).build_transaction({"gas": 1, "gasPrice": 1})["data"]
+        ).build_transaction(get_empty_tx_params())["data"]
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply gnosis/safe/multi_send.py:313]: I think in some previous version of Web3 `0` was taken as empty, so I decided to go with `1`

File: gnosis/eth/utils.py [hunk 10]
```diff
@@ -9,6 +9,17 @@
...
+from web3.types import TxParams, Wei
+
+
+def get_empty_tx_params() -> TxParams:
+    """
+    :return: Empty tx params, so calls like `build_transaction` don't call the RPC trying to get information
+    """
+    return {
+        "gas": Wei(1),
+        "gasPrice": Wei(1),
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply gnosis/eth/utils.py:21]: I think `1` is better due to it cannot be evaluated as empty

File: gnosis/safe/addresses.py [hunk 11]
```diff
@@ -5,8 +5,14 @@
 """
 from typing import Dict, List, Tuple
 
+from eth_typing import ChecksumAddress
+
 from gnosis.eth import EthereumNetwork
 
+SAFE_SIMULATE_TX_ACCESSOR_ADDRESS: ChecksumAddress = (
+    "0x3d4BA2E0884aa488718476ca2FB8Efc291A46199"
+)
```
[the remark this answers is not in the document]
Comment (Uxio0) [reply gnosis/safe/addresses.py:14]: Sure, but we should do it in another PR with other hardcoded addresses

File: gnosis/safe/safe.py (hunk 7 above)
Comment (Uxio0) [reply gnosis/safe/safe.py:70]: To separate the logic of creatign Safes into another class. It's just for organization and maybe in the future to split Safe creation logic from Safe class

**Review by moisses89 — APPROVED** [review]

**Pull request merged**